### PR TITLE
bump python from 3.11 to 3.12

### PR DESCRIPTION
**Pull request by damonmcc**
active support for 3.11 [ended](https://endoflife.date/python) ~3 months ago

## steps taken
- change the `PYTHON_VERSION` variable which is used by our docker files
- run action to build our `base` image using these changes [here](https://github.com/NYCPlanning/data-engineering/actions/runs/9668571351)
  - !! after doing this, every branch will be using this base image even before this is merged to `main` !!
- re-run docker image builds in this PR's tests [here](https://github.com/NYCPlanning/data-engineering/actions/runs/9668446567/job/26673944389#step:4:286) to use new base image
  - old base image [tag](https://github.com/NYCPlanning/data-engineering/actions/runs/9668446567/job/26672648518#step:4:286) and new base image [tag](https://github.com/NYCPlanning/data-engineering/actions/runs/9668446567/job/26673944389#step:4:286)
- compile python packages using this branch
- run all builds [here](https://github.com/NYCPlanning/data-engineering/actions/runs/9682659915) using new python and package versions (in-progress)

### CODE REVIEW

File: python/constraints.txt [hunk 1]
```diff
@@ -165,14 +159,12 @@ et-xmlfile==1.1.0
...
 executing==2.0.1
     # via stack-data
-faker==25.9.1
+faker==25.9.2
     # via -r python/requirements.in
 filelock==3.15.4
     # via
     #   gdown
     #   virtualenv
-fiona==1.9.6
```
Comment (fvankrieken) [inline python/constraints.txt:174]: Intriguing

File: python/constraints.txt [hunk 2]
```diff
@@ -192,7 +184,7 @@ geographiclib==2.0
     # via geopy
 geojson==3.1.0
     # via leafmap
-geopandas==0.14.4
+geopandas==1.0.0
```
Comment (fvankrieken) [inline python/constraints.txt:187]: Ah - unrelated to python version, this was released yesterday!

@sf-dcp big news

File: python/constraints.txt (hunk 1 above)
Comment (damonmcc) [reply python/constraints.txt:174]: this is probably due `geopandas==1.0.0`? and good news? lol

File: python/constraints.txt (hunk 2 above)
Comment (sf-dcp) [reply python/constraints.txt:187]: that's awesome!

File: python/constraints.txt (hunk 1 above)
Comment (sf-dcp) [reply python/constraints.txt:174]: my guess is geopandas changing their default engine for geo files from `fiona` to `pyogrio`...

Comment (sf-dcp) [reply python/constraints.txt:174]: yep,[ it changed](https://geopandas.org/en/latest/docs/user_guide/fiona_to_pyogrio.html#) and it's supposed to be faster.

image

**Comment by sf-dcp** [comment]
is this PR supposed to be a draft? Otherwise ready to approve!

**Review by fvankrieken — APPROVED** [review]

File: products/template/build_scripts/export.py
Comment (damonmcc) [inline products/template/build_scripts/export.py:1]: these changes were due to errors caused by the `geopandas` upgrade

errors [here](https://github.com/NYCPlanning/data-engineering/actions/runs/9669226221/job/26675241974#step:12:47) and [here](https://github.com/NYCPlanning/data-engineering/actions/runs/9669226221/job/26675241974#step:12:47)

**Comment by damonmcc** [comment]
gonna merge to unblock #952

**Pull request merged**